satpy/readers/yaml_reader.py

```diff
@@ -386,11 +391,12 @@ def new_filehandler_instances(self, filetype_info, filename_items):
             try:
                 req_fh = self.find_required_filehandlers(requirements,
                                                          filename_info)
-            except RuntimeError:
-                logger.warning("Can't find requirements for %s", filename)
+            except KeyError as req:
+                msg = "Missing requirement {} for {}".format(req, filename)
```

Comment:
Doesn't this put a string version of the exception `req` inside the error message? Should this be the specific requirement that is missing?

Reply:
The string version of the ``KeyError`` already contains the missing requirement. Using the string version is recommended (as far as I know). The actual problem is using ``RuntimeError.message`` because this attribute was removed in Python 3. I tested this with Python 2 :see_no_evil:

Reply:
That's fine by me I guess. If you have an example of what the final error message looks like that would be appreciated.

Reply:
The warnings would be :
1) ``UserWarning: No handler for reading requirement 'HRIT_PRO' for H-000-MSG4__-MSG4________-IR_108___-000006___-201809050900-__`` if none of the provided scans is complete
2) ``UserWarning: No matching requirement file of type HRIT_PRO for H-000-MSG3__-MSG3________-IR_108___-000008___-201801181200-__`` if at least one of the provided scans is complete and at least one is incomplete